cpp/src/gandiva/cache.h

```diff
@@ -26,7 +28,12 @@ namespace gandiva {
 template <class KeyType, typename ValueType>
 class Cache {
  public:
-  explicit Cache(size_t capacity = CACHE_SIZE) : cache_(capacity) {}
+  explicit Cache(size_t capacity) : cache_(capacity) {
+    std::cout << "Creating gandiva cache with capacity: " << capacity << std::endl;
```

Comment:
Two things:
1) I don't think it's ok to unconditionally log such informational messages. Please use e.g. `ARROW_LOG(INFO)`.
2) Please avoid, if at all possible, exposing such IO in header files. Better define the `Cache` constructor in `cache.cc`.